demos/object_detection_demo_ssd_async/main.cpp

```diff
@@ -235,102 +298,88 @@ int main(int argc, char *argv[]) {
                 data[1] = static_cast<float>(netInputWidth);  // width
                 data[2] = 1;
             };
-            setImgInfoBlob(async_infer_request_curr);
-            setImgInfoBlob(async_infer_request_next);
+
+            for (const InferRequest::Ptr &requestPtr : userSpecifiedInferRequests) {
+                setImgInfoBlob(requestPtr);
+            }
+            setImgInfoBlob(minLatencyInferRequest);
         }
         // -----------------------------------------------------------------------------------------------------
 
-        // --------------------------- 6. Do inference ---------------------------------------------------------
-        slog::info << "Start inference " << slog::endl;
+        // --------------------------- 6. Init variables -------------------------------------------------------
+        typedef std::chrono::duration<double, std::chrono::milliseconds::period> Ms;
+        typedef std::chrono::duration<double, std::chrono::seconds::period> Sec;
+        
+        struct RequestResult {
+            cv::Mat frame;
+            std::unique_ptr<LockedMemory<const void>> output;
+            std::chrono::steady_clock::time_point startTime;
+            bool isSameMode;
+        };
+
+        struct ModeInfo {
+           int framesCount = 0;
+           std::chrono::steady_clock::duration latencySum;
+           std::chrono::steady_clock::time_point lastStartTime;
+           std::chrono::steady_clock::time_point lastEndTime;
+        };
+
+        bool isUserSpecifiedMode = true;  // execution always starts in USER_SPECIFIED mode
+
+        auto totalT0 = std::chrono::steady_clock::now();
+
+        std::deque<InferRequest::Ptr> emptyRequests;
+        if (isUserSpecifiedMode) {
+            emptyRequests.assign(userSpecifiedInferRequests.begin(), userSpecifiedInferRequests.end());
+        } else emptyRequests = {minLatencyInferRequest};
+
+        std::unordered_map<int, RequestResult> completedRequestResults;
+        int nextFrameId = 0;
+        int nextFrameIdToShow = 0;
+        std::exception_ptr callbackException = nullptr;
+        std::mutex mutex;
+        std::condition_variable condVar;
+        std::map<bool, ModeInfo> modeInfo = {{true, ModeInfo()}, {false, ModeInfo()}};
```

Comment:
The usage of `bool` for the execution mode makes the code hard to understand, because it's not obvious which is `true` and which is `false`. I would suggest introducing an `enum class ExecutionMode` instead.